classes/log/store.php

```diff
@@ -118,11 +146,18 @@ public function process_events(array $events) {
             $eventbatches = array_chunk($translatorevents, $maxbatchsize);
         }
 
+		$translator_event = new Event();
+		$translator_event_read_return = @$translator_event->read(array());
```

Comment:
To catch the dispatch status and assign it to the correct event id, I had to parse the returned structure (see under). The event id is inside an associative array which key is "http://lrs.learninglocker.net/define/extensions/moodle_logstore_standard_log". Instead of hardcoding the key here too, I read it from the Event, faking a read to get the context_ext_key which contains this key. Since I was afraid the key could change, invalidating the structure parsing, I decide to go along this path.